Quick note on the two failing checks — both are pre-existing on `master`, not introduced by this PR:

**1. Code Quality Scan (`build (ubuntu-latest)`) — `F841` on `optimization.py:1168`**

`uvx ruff check .` reports:
> `src/emhass/optimization.py:1168:9: F841 Local variable 'p_grid_neg' is assigned to but never used`

This is a dead local assignment in `_add_battery_constraints`: `p_grid_neg = self.vars["p_grid_neg"]` is never referenced in the body of that function. Pre-existing since `d50aacd4` (2026-01-20). The Code Quality Scan workflow has been failing on every push to `master` for several days (last 5 runs all `failure`).

**Severity: lint-only.** `p_grid_neg` itself is correctly used everywhere it actually matters — variable definition (`optimization.py:796-798`), cost function (`L899-949`), balance equations (`L1008-1057`, `L1032/1041/1049`), max-export constraint (`L1057`), and result extraction (`L2298`). The grid-export constraints live in `_add_balance_constraints`, not here, so the optimization is unaffected. The fix is to drop the line.

**2. CodeCov (`run`) — `SonarQube Scan: Project not found ... 'SONAR_TOKEN'`**

GitHub Actions does not expose repository secrets to workflows triggered from forks, so `${{ secrets.SONAR_TOKEN }}` is empty here, and Sonar fails with `ERROR Project not found`. This is expected for any external-contributor PR; not a code issue.

Codecov itself is green: `codecov/patch` and `codecov/project` both ✅, coverage `+0.01%` (all new lines covered). Python test matrix (ubuntu/macos/windows), CodeQL, Docker builds, OSV scans — all green.

Happy to open a separate cleanup PR for the `F841` if useful, or leave it for whoever next touches `_add_battery_constraints`.

--- a/src/emhass/utils.py
+++ b/src/emhass/utils.py
@@ -1233,6 +1233,17 @@ async def treat_runtimeparams(
             weather_forecast_cache_only = runtimeparams["weather_forecast_cache_only"]
         params["passed_data"]["weather_forecast_cache_only"] = weather_forecast_cache_only
 
+        # Param to bypass PV-feedback mixing during curtailment events (#818)
+        if "ignore_pv_feedback_during_curtailment" not in runtimeparams.keys():
+            ignore_pv_feedback_during_curtailment = False
+        else:
+            ignore_pv_feedback_during_curtailment = bool(
+                runtimeparams["ignore_pv_feedback_during_curtailment"]
+            )
+        params["passed_data"]["ignore_pv_feedback_during_curtailment"] = (
+            ignore_pv_feedback_during_curtailment
+        )
+
         # A condition to manually save entity data under data_path/entities after optimization
         if "entity_save" not in runtimeparams.keys():
             entity_save = ""

--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -1281,6 +1281,52 @@ class TestUtils(unittest.IsolatedAsyncioTestCase):
         self.assertEqual(params_out["passed_data"]["soc_init"], 0.95)
         self.assertEqual(params_out["passed_data"]["soc_final"], 0.6)
 
+    async def test_treat_runtimeparams_ignore_pv_feedback_during_curtailment(self):
+        """Wiring for ignore_pv_feedback_during_curtailment runtime flag (#818).
+
+        The read site in forecast.py reads from params["passed_data"]; this
+        test pins the runtime → passed_data path for the four realistic input
+        shapes: missing key (default False), JSON bool true, string "true",
+        string "false". The string cases document the bool() coerce behaviour.
+        """
+        params = await TestUtils.get_test_params()
+        params_json = orjson.dumps(params).decode("utf-8")
+        retrieve_hass_conf, optim_conf, plant_conf = utils.get_yaml_parse(params_json, logger)
+
+        async def run(runtimeparams_dict):
+            runtimeparams_json = orjson.dumps(runtimeparams_dict).decode("utf-8")
+            params_out, _, _, _ = await treat_runtimeparams(
+                runtimeparams_json,
+                params_json,
+                retrieve_hass_conf,
+                optim_conf,
+                plant_conf,
+                "naive-mpc-optim",
+                logger,
+                emhass_conf,
+            )
+            return orjson.loads(params_out)
+
+        # Case 1: key absent -> default False
+        out = await run({"prediction_horizon": 10})
+        self.assertIs(out["passed_data"]["ignore_pv_feedback_during_curtailment"], False)
+
+        # Case 2: JSON bool true -> True
+        out = await run({"prediction_horizon": 10, "ignore_pv_feedback_during_curtailment": True})
+        self.assertIs(out["passed_data"]["ignore_pv_feedback_during_curtailment"], True)
+
+        # Case 3: string "true" -> bool() coerce -> True
+        out = await run({"prediction_horizon": 10, "ignore_pv_feedback_during_curtailment": "true"})
+        self.assertIs(out["passed_data"]["ignore_pv_feedback_during_curtailment"], True)
+
+        # Case 4: string "false" -> bool() coerce -> True (Python bool("false") is True)
+        # Documents the known limitation of bool() on non-empty strings;
+        # JSON bool transport is the supported shape.
+        out = await run(
+            {"prediction_horizon": 10, "ignore_pv_feedback_during_curtailment": "false"}
+        )
+        self.assertIs(out["passed_data"]["ignore_pv_feedback_during_curtailment"], True)
+
     def test_param_to_config(self):
         """Test converting built params back to a flat config dictionary and masking secrets."""
         # Create a mock parameter dictionary with the required categories

